openshift/machine-config-operator
# Bug 1835233: Add support for configuring registries at install

Fixes #1835233 (https://bugzilla.redhat.com/show_bug.cgi?id=1835233)
Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>



**- What I did**
Add support to bootstrap to configure the registries in the
cluster-wide Image CR. Ensure that the MCC can consume the
Image CR and generate /etc/containers/registries.conf correctly.

**- How to verify it**
1. Run `openshift-install create manifests`
2. Add the following Image CR yaml to the manifests directory created from step 1
```
apiVersion: config.openshift.io/v1
kind: Image
metadata:
  creationTimestamp: null
  name: cluster
spec:
  registrySources:
    insecureRegistries:
    - blah.io
    - foo.io
```
3. Run `openshift-install create cluster`
4. Watch to ensure that the cluster was successfully installed and check the `/etc/containers/registries.conf` file on all the nodes to see that it has the configuration from the Image CR in step 2.

**- Description for the changelog**

Add day 1 support for registries configured in the cluster-wide Image CR
